.github/workflows/pr-test-lint.yml

```diff
@@ -90,10 +90,10 @@ jobs:
 
   go-auto-approve:
     runs-on: ubuntu-latest
-    needs: go-test
+    needs: [go-test, go-bench, golangci-lint, go-mod]
     # For now, we only auto approve and merge go dep PRs because we have tests for this in place.
     # https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#startswith
-    if: ${{ startsWith(github.ref, 'refs/heads/dependabot/go_modules') && github.actor == 'dependabot[bot]' && needs.go-test.outputs.outcome == 'success' }}
+    if: ${{ startsWith(github.ref, 'refs/heads/dependabot/go_modules') && github.actor == 'dependabot[bot]' }}
```

Comment:
`needs` implies success: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds